lib/Dialect/LWE/Conversions/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -467,6 +497,52 @@ struct ConvertRMul : public OpConversionPattern<RMulOp> {
   }
 };
 
+struct ConvertRMulPlain : public OpConversionPattern<RMulOp_Plain> {
+  ConvertRMulPlain(mlir::MLIRContext *context)
+      : OpConversionPattern<RMulOp_Plain>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  // verify num elements, verify order of ciphertext-plaintext
+  LogicalResult matchAndRewrite(
+      RMulOp_Plain op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto x = adaptor.getLhs();
+    auto xT = cast<RankedTensorType>(x.getType());
+    auto y = adaptor.getRhs();
+    auto yT = cast<RankedTensorType>(y.getType());
+
+    if (xT.getNumElements() != 2 || yT.getNumElements() != 1) {
```

Comment:
If the plaintext op can be in either operand, you may want to add a step before this that checks the types of `op.getLhs()` and `op.getRhs()` and then swaps `x` and `y` appropriately.

Or we could:

- make the tablegen force the LHS to be ciphertext and RHS to be plaintext
- add a general canonicalization rule moves the plaintext to the RHS. 

The lowerings to LWE may already ensure this, but new lowerings to LWE may not be aware of this requirement.